changelog.d/pr-10684.md

```diff
@@ -0,0 +1,74 @@
+---
+synopsis: "A project target is required for the REPL command"
+packages: [cabal-install]
+prs: 10684
+issues: 10527
+---
+
+With a project, the REPL command requires a target. If one is not given then a
+message is shown explaining this and naming the project if the `--project-file`
+option was given (but not when the default 'cabal.project' project name is used
+implicitly).  We're not yet able to list project targets so in the meantime, the
+messages lists the packages of the project.
+
+* When the implicit default `cabal.project` is used:
+
+```
+$ cat cabal.project
+packages: pkg-one pkg-two
+
+$ cabal repl
+Error: [Cabal-7076]
+With a project, the REPL command requires a single target. The packages in this project are:
+ - pkg-one
+ - pkg-two
```

Comment:
I'm not sure it's a good idea to silently blur the distinction between targets and packages like this. Something like
```suggestion
With a project, the REPL command requires a single target. You can find possible targets by inspecting the packages in this project:
 - pkg-one
 - pkg-two
```

Reply:
I've changed the message to be more specific. It says "pick a single [package:][ctype:]component as target for the REPL command". This is the [component target syntax](https://cabal.readthedocs.io/en/latest/cabal-commands.html#target-forms) from the "Target Forms" section of the docs.